drivers/SAI.cpp

```diff
@@ -38,8 +38,11 @@ SAI::SAI(PinName mclk, PinName bclk, PinName wclk, PinName sd,
 
     init.format = *fmt;
 
-    if (sai_init(&_sai, &init) != SAI_RESULT_OK) {
+    _status = sai_init(&_sai, &init);
+
+    if (_status != SAI_RESULT_OK) {
         // it failed :o
+        printf("SAI init failed with: %d\r\n", _status);
```

Comment:
Is this debug print needed? 
Status now can be checked using `SAI::status` member function.